lib/scala/kg/src/main/scala/formats/CsvReader.scala

```diff
@@ -0,0 +1,51 @@
+package formats
+
+import com.github.tototoshi.csv.CSVReader
+import java.nio.file.Path
+import java.io.BufferedInputStream
+import java.io.FileInputStream
+import java.io.InputStream
+import java.io.InputStreamReader
+import org.apache.commons.compress.compressors.CompressorStreamFactory
+import org.apache.commons.compress.compressors.CompressorException
+import java.io.Reader
+import com.github.tototoshi.csv.CSVFormat
+import org.apache.commons.lang3.StringUtils
+
+abstract class CsvReader[T](protected val csvReader: CSVReader) extends AutoCloseable with Iterable[T] {
```

Comment:
Nice abstraction, this is very appropriate given the (hopefully temporary) legacy consideration.